collect_app/src/main/java/org/odk/collect/android/tasks/FormLoaderTask.java

```diff
@@ -220,29 +218,13 @@ protected FECWrapper doInBackground(String... path) {
         return data;
     }
 
-    private FormDef createFormDefFromCacheOrXml(String formPath, File formXml) {
-        final String formHash = FileUtils.getMd5Hash(formXml);
-
+    private FormDef createFormDefFromCacheOrXml(File formXml) {
         publishProgress(
                 Collect.getInstance().getString(R.string.survey_loading_reading_form_message));
 
-        final File cachedForm = new File(Collect.CACHE_PATH + File.separator + formHash + ".formdef");
-        if (cachedForm.exists()) {
-            Timber.i("Attempting to load %s from cached file: %s.",
-                    formXml.getName(), cachedForm.getAbsolutePath());
-            final long start = System.currentTimeMillis();
-            final FormDef deserializedFormDef = deserializeFormDef(cachedForm);
-            if (deserializedFormDef != null) {
-                Timber.i("Loaded in %.3f seconds.", (System.currentTimeMillis() - start) / 1000F);
-
-                return deserializedFormDef;
-            }
-
-            // An error occurred with deserialization. Remove the file, and make a
-            // new .formdef from xml.
-            Timber.w("Deserialization FAILED! Deleting cache file: %s",
-                    cachedForm.getAbsolutePath());
-            cachedForm.delete();
+        final FormDef formDefFromCache = new FormDefCache().readCache(formXml);
```

Comment:
`new FormDefCache().readCache(formXml)` can be replaced with `FormDefCache.readCache(formXml)`

Reply:
I don’t know how I missed that. IDEA is calling it out to me, just as it probably did to you. Thanks. @srv-twry, you’ll take care of it?

Reply:
@srv-twry Do you agree that this should be changed?

Reply:
Yeah.